src/relax/op/op.cc

```diff
@@ -292,7 +349,10 @@ RELAY_REGISTER_OP("relax.invoke_closure")
     .set_num_inputs(2)
     .add_argument("closure", "Expr", "The VMClosure.")
     .add_argument("args", "Tuple", "The captured variables.")
-    .set_attr<FInferStructInfo>("FInferStructInfo", InferStructInfoInvokeClosure);
+    .set_attr<FInferStructInfo>("FInferStructInfo", InferStructInfoInvokeClosure)
+    // TODO: This might be another case where we would want a macro or even use an attr.
+    // It may depend on the particulars of the closure
+    .set_attr<Bool>("FPurity", Bool(false));
```

Comment:
Here is an example where using a simple boolean for indicating the purity of an op might not work out. I don't know if there is any way to figure out whether a given VM closure is pure or not. 

Alternatively, if we take the view that we should not track purity past a certain point in compilation (e.g., by the time we get rid of dataflow blocks), maybe this is less of an issue.

Reply:
this concern is resolved now with ForcePure and RemoveTrackingPurity, right?

Reply:
Sort of, it's resolved with `invoke_pure_closure`. See the changes to `LambdaLifting`: it reasons about whether the function that was turned into a closure is pure.